docs/guides/js-waku/message-encryption.md

```diff
@@ -0,0 +1,218 @@
+	publicKey: publicKey, // ECIES public key for encrypting messages
+});
+
+// Send the message using Light Push
+await node.lightPush.send(encoder, { payload });
+```
+
+To decrypt the messages you receive, create an `ECIES` message `decoder` with the private key and process the messages as usual:
+
+```js
+import { createDecoder } from "@waku/message-encryption/ecies";
+
+// Create an ECIES message decoder
+const decoder = createDecoder(contentTopic, privateKey);
+
+// Receive messages from a Filter subscription
+const subscription = await node.filter.createSubscription();
+await subscription.subscribe([decoder], callback);
+
+// Retrieve messages from Store peers
+await node.store.queryWithOrderedCallback([decoder], callback);
+```
+
+## Signing encrypted messages
+
+Message signing helps in proving the authenticity of received messages. By attaching a signature to a message, you can verify its origin and integrity with absolute certainty.
+
+The `sigPrivKey` option allows the `Symmetric` and `ECIES` message `encoders` to sign the message before encryption using an `ECDSA` private key:
+
+```js
+import { generatePrivateKey } from "@waku/message-encryption";
+import { createEncoder as createSymmetricEncoder } from "@waku/message-encryption/symmetric";
+import { createEncoder as createECIESEncoder } from "@waku/message-encryption/ecies";
+
+// Generate a random private key for signing messages
+const sigPrivKey = generatePrivateKey();
+
+// Create a symmetric encoder that signs messages
+const symmetricEncoder = createSymmetricEncoder({
+	contentTopic: contentTopic, // message content topic
+	symKey: symKey, // symmetric key for encrypting messages
+	sigPrivKey: sigPrivKey, // private key for signing messages before encryption
+});
+
+// Create an ECIES encoder that signs messages
+const ECIESEncoder = createECIESEncoder({
+	contentTopic: contentTopic, // message content topic
+	publicKey: publicKey, // ECIES public key for encrypting messages
+	sigPrivKey: sigPrivKey, // private key for signing messages before encryption
+});
+
+// Send and receive your messages as usual with Light Push and Filter
+await subscription.subscribe([symmetricEncoder], callback);
+await node.lightPush.send(symmetricEncoder, { payload });
+
+await subscription.subscribe([ECIESEncoder], callback);
+await node.lightPush.send(ECIESEncoder, { payload });
+```
+
+You can extract the `signature` and its public key (`signaturePublicKey`) from the [DecodedMessage](https://js.waku.org/classes/_waku_message_encryption.DecodedMessage.html) and compare it with the expected public key to verify the message:
+
+```js
+// Generate a random private key for signing messages
+const sigPrivKey = generatePrivateKey();
+
+// Generate a public key from the private key for verifying signatures
+const sigPubKey = getPublicKey(sigPrivKey);
+
+// Create an encoder that signs messages
+const encoder = createEncoder({
+	contentTopic: contentTopic,
+	symKey: symKey,
+	sigPrivKey: sigPrivKey,
+});
```

Comment:
I think for this example to talk to the reader, we need to introduce Alice/Bob or local/remote.

Reply:
![Screenshot 2023-12-08 at 13 43 54](https://github.com/waku-org/docs.waku.org/assets/47832826/4b98d264-7851-49c0-b2f3-a3d07ecb46b0)

went with the Alice/Bob reference